SwrveSDK/src/main/java/com/swrve/sdk/locationcampaigns/LocationCampaignFilter.java

```diff
@@ -0,0 +1,72 @@
+package com.swrve.sdk.locationcampaigns;
+
+import android.Manifest;
+import android.content.Context;
+import android.os.PowerManager;
+import android.support.annotation.RequiresPermission;
+import android.util.Log;
+
+import com.plotprojects.retail.android.FilterableNotification;
+import com.plotprojects.retail.android.NotificationFilterReceiver;
+import com.swrve.sdk.SwrveSDKBase;
+import com.swrve.sdk.locationcampaigns.model.LocationCampaignPayload;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LocationCampaignFilter extends NotificationFilterReceiver {
+
+    private static final String LOG_TAG = "SwrveLocation";
+
+    @RequiresPermission(Manifest.permission.WAKE_LOCK)
+    @Override
+    public List<FilterableNotification> filterNotifications(List<FilterableNotification> notifications) {
+
+        Log.d(LOG_TAG, "LocationCampaignNotificationFilter. Got " + notifications.size() + " notifications from plot projects.");
+        PowerManager mgr = (PowerManager) getSystemService(Context.POWER_SERVICE);
+        PowerManager.WakeLock wakeLock = mgr.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, "swrve_wakelock");
+        try {
+            wakeLock.acquire();
+
+            // todo dom: presumption here that we don't need to refresh location campaigns
```

Comment:
For FA we're just relying on the app being started and fresh campaigns being pulled down. This will definitely change for GA.